page_parser.py

```diff
@@ -0,0 +1,21 @@
+"""
+A module to store functions related to parsing the HTML of the Ringing Room tower pages to find
+things like the load-balanced URL of the socket-io server.
+"""
+
+import os
+import requests
+
+def get_load_balancing_url(tower_id, http_server_url):
+    """
+    Get the URL of the socket server which (since the addition of load balancing) is not
+    necessarily the same as the URL of the http server that people will put into their browser URL
+    bars.
+    """
+
+    html = requests.get(os.path.join(http_server_url, str(tower_id), 'x')).text
+
+    string_that_starts_with_url = html[html.index("server_ip") + len('server_ip: "'):]
```

Comment:
This is an implementation detail of ringing room, so may change without warning.

1) Worth commenting something like:
```python
# Trying to extract the following line in the rendered html:
# server_ip: "{{server_ip}}"
# See https://github.com/lelandpaul/virtual-ringing-room/blob/ec00927ca57ab94fa2ff6a978ffaff707ab23a57/app/templates/ringing_room.html#L46
```
2) Does `.index()` on string raise an exception if it doesn't find the substring? If not we should raise one here.
3) I found it a little hard to parse this line all at once, I think a variable like
```
 ip_start_index = html.index('server_ip') + len('server_ip: "')
```
would help.

Reply:
`.index()` throws a `ValueError` if the substring cannot be found.  I've made the other changes too :).